### PR TITLE
Make routing documentation more coherent

### PR DESCRIPTION
**Pull request by sedubois**
- add missing documentation (e.g `as` is undocumented in declarative <Link>)
- remove documentation redundancies
- hide away the inline examples in "details" snippet

The improvements are not perfect, but if they are good enough, suggest to merge and refine in later PR.

- [ ] update TOC (how?)

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -362,7 +361,7 @@ Router.onAppUpdated = (nextUrl) => {
 
 Shallow routig allows you to change the URL without running `getInitialProps`. You'll receive the updated `pathname` and the `query` via the `url` prop of the same page that's loaded, without losing state.
```
Comment (cdock1029) [inline readme.md]: routig => **routing**

Comment (timneutkens) [inline readme.md]: Someone fixed this in another PR already 😄

**Comment by timneutkens** [comment]
On generating the TOC:
https://github.com/zeit/next.js/blame/master/readme.md#L11

It uses `doctoc`

1. `npm install -g doctoc`
2. `doctoc`

Maybe we should add doctoc as a devdep.

**Comment by timneutkens** [comment]
Lets merge this after https://github.com/zeit/next.js/pull/1256

**Pull request closed without merging**